### PR TITLE
 Upgrade Jedi to 0.12.0

### PR DESCRIPTION
**Pull request by edelvalle**
Use Jedi directly inside the sublime Python interpreter without break the interface.
Notice this is a suggestion...

Using Environments:
- I infer the virtualenv from the python interpreter path.
- and the python interpreter path from the virtualenv.
- if that does not work I fallback to the default interpreter.
- to support `extra_packages` the I use the `sys_path` parameter of the `jedi.Script` constructor.
- I removed all the threading stuff, and use `set_timeout_async` in sublime 3 and `set_timeout` in sublime 3. So it is non-blocking for sublime 3, and blocking for sublime 2. I think this is a way to make people also upgrade their sublimes.... :laughing: 

@srusskih give me your opinion 
Cheers,
Eddy

### CODE REVIEW

File: sublime_jedi/utils.py [hunk 1]
```diff
@@ -1,169 +1,81 @@
...
 
-            with self.wait_lock:
-                self.waiting[data['uuid']] = callback
+    def __init__(self, view):
+        self.window_id = view.window().id()
+        self.process = self._start_process(get_settings(view))
 
-            if not isinstance(data, str):
-                data = json.dumps(data)
+    def _start_process(self, settings):
```
Comment (srusskih) [inline sublime_jedi/utils.py]: I think we no more start process here, looks like daemon configuration

File: sublime_jedi/utils.py [hunk 2]
```diff
@@ -221,24 +130,17 @@ def get_settings(view):
...
-    python_interpreter = get_settings_param(view, 'python_interpreter_path')
-
-    if not python_interpreter:
-        python_interpreter = get_settings_param(view, 'python_interpreter',
-                                                'python')
-    else:
-        warnings.warn('`python_interpreter_path` parameter is deprecated.'
-                      'Please, use `python_interpreter` instead.',
-                      DeprecationWarning)
+    python_virtualenv = get_settings_param(view, 'python_virtualenv', None)
```
Comment (srusskih) [inline sublime_jedi/utils.py]: I think we should add new option to README

File: sublime_jedi/utils.py (hunk 1 above)
Comment (edelvalle) [reply sublime_jedi/utils.py]: Yes, will rename it.

**Comment by edelvalle** [comment]
Sure, I will do so using the Python 2 API for that... `thread.start_new_thread`, is that ok?

**Comment by srusskih** [comment]
Sounds good

**Comment by edelvalle** [comment]
@srusskih Ready to go!

File: sublime_jedi/utils.py [hunk 3]
```diff
@@ -131,6 +134,8 @@ def get_settings(view):
     :rtype: dict
     """
     python_virtualenv = get_settings_param(view, 'python_virtualenv', None)
+    if python_virtualenv:
+        python_virtualenv = expand_path(view, python_virtualenv)
```
Comment (srusskih) [inline sublime_jedi/utils.py]: 👍

**Pull request merged**